avro-converter/src/main/java/io/confluent/copycat/avro/AvroData.java

```diff
@@ -0,0 +1,1073 @@
+  static {
+    SIMPLE_AVRO_SCHEMA_TYPES.put(Schema.Type.INT32, Arrays.asList((Class) Integer.class));
+    SIMPLE_AVRO_SCHEMA_TYPES.put(Schema.Type.INT64, Arrays.asList((Class) Long.class));
+    SIMPLE_AVRO_SCHEMA_TYPES.put(Schema.Type.FLOAT32, Arrays.asList((Class) Float.class));
+    SIMPLE_AVRO_SCHEMA_TYPES.put(Schema.Type.FLOAT64, Arrays.asList((Class) Double.class));
+    SIMPLE_AVRO_SCHEMA_TYPES.put(Schema.Type.BOOLEAN, Arrays.asList((Class) Boolean.class));
+    SIMPLE_AVRO_SCHEMA_TYPES.put(Schema.Type.STRING, Arrays.asList((Class) CharSequence.class));
+    SIMPLE_AVRO_SCHEMA_TYPES.put(
+        Schema.Type.BYTES,
+        Arrays.asList((Class) ByteBuffer.class, (Class) byte[].class, (Class) GenericFixed.class));
+    SIMPLE_AVRO_SCHEMA_TYPES.put(Schema.Type.ARRAY, Arrays.asList((Class) Collection.class));
+    SIMPLE_AVRO_SCHEMA_TYPES.put(Schema.Type.MAP, Arrays.asList((Class) Map.class));
+  }
+
+  private static final Map<Schema.Type, org.apache.avro.Schema.Type> COPYCAT_TYPES_TO_AVRO_TYPES
+      = new HashMap<>();
+  static {
+    COPYCAT_TYPES_TO_AVRO_TYPES.put(Schema.Type.INT32, org.apache.avro.Schema.Type.INT);
+    COPYCAT_TYPES_TO_AVRO_TYPES.put(Schema.Type.INT64, org.apache.avro.Schema.Type.LONG);
+    COPYCAT_TYPES_TO_AVRO_TYPES.put(Schema.Type.FLOAT32, org.apache.avro.Schema.Type.FLOAT);
+    COPYCAT_TYPES_TO_AVRO_TYPES.put(Schema.Type.FLOAT64, org.apache.avro.Schema.Type.DOUBLE);
+    COPYCAT_TYPES_TO_AVRO_TYPES.put(Schema.Type.BOOLEAN, org.apache.avro.Schema.Type.BOOLEAN);
+    COPYCAT_TYPES_TO_AVRO_TYPES.put(Schema.Type.STRING, org.apache.avro.Schema.Type.STRING);
+    COPYCAT_TYPES_TO_AVRO_TYPES.put(Schema.Type.BYTES, org.apache.avro.Schema.Type.BYTES);
+    COPYCAT_TYPES_TO_AVRO_TYPES.put(Schema.Type.ARRAY, org.apache.avro.Schema.Type.ARRAY);
+    COPYCAT_TYPES_TO_AVRO_TYPES.put(Schema.Type.MAP, org.apache.avro.Schema.Type.MAP);
+  }
+
+
+  private static final String ANYTHING_SCHEMA_BOOLEAN_FIELD = "boolean";
+  private static final String ANYTHING_SCHEMA_BYTES_FIELD = "bytes";
+  private static final String ANYTHING_SCHEMA_DOUBLE_FIELD = "double";
+  private static final String ANYTHING_SCHEMA_FLOAT_FIELD = "float";
+  private static final String ANYTHING_SCHEMA_INT_FIELD = "int";
+  private static final String ANYTHING_SCHEMA_LONG_FIELD = "long";
+  private static final String ANYTHING_SCHEMA_STRING_FIELD = "string";
+  private static final String ANYTHING_SCHEMA_ARRAY_FIELD = "array";
+  private static final String ANYTHING_SCHEMA_MAP_FIELD = "map";
+
+  public static final org.apache.avro.Schema ANYTHING_SCHEMA_MAP_ELEMENT;
+  public static final org.apache.avro.Schema ANYTHING_SCHEMA;
+  static {
+    // Intuitively this should be a union schema. However, unions can't be named in Avro and this
+    // is a self-referencing type, so we need to use a format in which we can name the entire schema
+
+    ANYTHING_SCHEMA =
```

Comment:
Specialized types are not part of anything? I feel like there's a good idea here that I'm somehow missing...